mlrun/projects/pipelines.py

```diff
@@ -573,7 +570,7 @@ def wait_for_completion(
 
     @staticmethod
     def get_state(run_id, project=None):
-        project_name = project.metadata.name if project else ""
+        project_name = project or ""
```

Comment:
seems like before the project param was of type project, and now its str, let's add type hinting, also... i see that this is a public method, meaning user can use it (which now we break backwards compatibility for the use) I would also keep the old behavior and log a warning about the previous type deprecation.

Reply:
adding type hint. this is also for inner use. see 3. in the description